test/tools/modlog/converter.js

```diff
@@ -0,0 +1,518 @@
+			const entry = {
+				action: 'UNITTEST',
+				roomID: 'development',
+				userid: 'annika',
+				autoconfirmedID: 'heartofetheria',
+				alts: ['googlegoddess', 'princessentrapta'],
+				ip: '127.0.0.1',
+				isGlobal: false,
+				loggedBy: 'yourmom',
+				note: 'Hey Adora~',
+				time: 1598212249944,
+			};
+			assert.strictEqual(
+				converter.rawifyLog(entry),
+				`[2020-08-23T19:50:49.944Z] (development) UNITTEST: [annika] ac: [heartofetheria] alts: [googlegoddess], [princessentrapta] [127.0.0.1] by yourmom: Hey Adora~\n`
+			);
+		});
+	});
+
+	describe('integration tests', () => {
+		it('should convert from SQLite to text', async () => {
+			const modlog = new ml.Modlog(':memory:', true);
+			const mlConverter = new converter.ModlogConverterSQLite('', '', modlog.database);
+
+			const entry = {
+				action: 'UNITTEST',
+				roomID: 'development',
+				userid: 'annika',
+				autoconfirmedID: 'heartofetheria',
+				alts: ['googlegoddess', 'princessentrapta'],
+				ip: '127.0.0.1',
+				isGlobal: false,
+				loggedBy: 'yourmom',
+				note: 'Write 1',
+				time: 1598212249944,
+			};
+			const writes = [];
+			writes.push(modlog.write('development', entry));
+			entry.time++;
+			entry.note = 'Write 2';
+			writes.push(modlog.write('development', entry));
+			entry.time++;
+			entry.note = 'Write 3';
+			writes.push(modlog.write('development', entry));
+			writes.push(modlog.write('development', {
+				action: 'GLOBAL UNITTEST',
+				roomID: 'development',
+				userid: 'annika',
+				autoconfirmedID: 'heartofetheria',
+				alts: ['googlegoddess', 'princessentrapta'],
+				ip: '127.0.0.1',
+				isGlobal: true,
+				loggedBy: 'yourmom',
+				note: 'Global test',
+				time: 1598212249947,
+			}));
+
+			await Promise.all(writes);
+			await mlConverter.toTxt();
+
+			assert.strictEqual(
+				mlConverter.isTesting.files.get('/modlog_development.txt'),
+				`[2020-08-23T19:50:49.944Z] (development) UNITTEST: [annika] ac: [heartofetheria] alts: [googlegoddess], [princessentrapta] [127.0.0.1] by yourmom: Write 1\n` +
+				`[2020-08-23T19:50:49.945Z] (development) UNITTEST: [annika] ac: [heartofetheria] alts: [googlegoddess], [princessentrapta] [127.0.0.1] by yourmom: Write 2\n` +
+				`[2020-08-23T19:50:49.946Z] (development) UNITTEST: [annika] ac: [heartofetheria] alts: [googlegoddess], [princessentrapta] [127.0.0.1] by yourmom: Write 3\n` +
+				`[2020-08-23T19:50:49.947Z] (development) GLOBAL UNITTEST: [annika] ac: [heartofetheria] alts: [googlegoddess], [princessentrapta] [127.0.0.1] by yourmom: Global test\n`
+			);
+
+			assert.strictEqual(
+				mlConverter.isTesting.files.get('/modlog_global.txt'),
+				`[2020-08-23T19:50:49.947Z] (development) GLOBAL UNITTEST: [annika] ac: [heartofetheria] alts: [googlegoddess], [princessentrapta] [127.0.0.1] by yourmom: Global test\n`
+			);
+		});
+
+		it('should convert from text to SQLite, including old-format lines and nonsense lines', async () => {
```

Comment:
Here's where I test it in the unit tests; I also ran the script on some autogenerated files (~10,000 entries) and it worked fine in both directions.